### PR TITLE
Deprecate `PosixException` in favor of `IOException`

### PR DESCRIPTION
**Pull request by basil**
See https://github.com/jenkinsci/jenkins/pull/6323#discussion_r817148584. As of #5560 and #6323, there is no longer a reason for a dedicated `PosixException` class. The generic `IOException` class suffices for all use cases. Therefore this PR deprecates `PosixException` and rewrites any usages of `PosixException` within Jenkins core to use `IOException` instead.

### Migration path

This is a change in behavior, albeit a minor one. Back in #5560 I did a survey of `PosixException` and found that it was used in [Copy Artifact](https://plugins.jenkins.io/copyartifact/), [CloudBees Backup Plugin](https://docs.cloudbees.com/docs/release-notes/latest/plugins/infradna-backup-plugin/), and [Operations Center Context Plugin](https://docs.cloudbees.com/docs/release-notes/latest/plugins/operations-center-context-plugin/). Today, it is also used in [JNR Posix API](https://plugins.jenkins.io/jnr-posix-api/).

- If the plugin needs to support both old versions of Jenkins (without this PR) and new versions of Jenkins (with this PR), then catch both `PosixException` and `IOException` and handle them both the same way. This is why I am merely deprecating `PosixException` in this PR rather than restricting or deleting it.
- If the plugin only needs to support new versions of Jenkins (with this PR), then replace all references to `PosixException` with references to `IOException`.

### Proposed changelog entries

Deprecate `PosixException` in favor of `IOException`.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
